service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/ControlledAzureDatabaseResource.java

```diff
@@ -148,52 +158,19 @@ public void addCreateSteps(
 
   @VisibleForTesting
   List<Step> getAddSteps(FlightBeanBag flightBeanBag) {
-    var getManagedIdentityStep =
-        switch (getAccessScope()) {
-          case ACCESS_SCOPE_SHARED -> new GetWorkspaceManagedIdentityStep(
-              flightBeanBag.getAzureConfig(),
-              flightBeanBag.getCrlService(),
-              getWorkspaceId(),
-              flightBeanBag.getResourceDao(),
-              getDatabaseOwner());
-
-          case ACCESS_SCOPE_PRIVATE -> new GetPetManagedIdentityStep(
-              flightBeanBag.getAzureConfig(),
-              flightBeanBag.getCrlService(),
-              flightBeanBag.getSamService(),
-              getAssignedUser().orElseThrow());
-        };
-
-    return List.of(
-        new GetAzureDatabaseStep(
+    var steps = new ArrayList<Step>();
+    steps.add(new ValidateDatabaseOwnerStep(this, flightBeanBag.getResourceDao()));
```

Comment:
validating the database owner used to be implicit in setting up the federated identity

Reply:
the pet managed identity will be created by sam if it does not exist